x/shield/keeper/purchase.go

```diff
@@ -176,6 +198,10 @@ func (k Keeper) RemoveExpiredPurchasesAndDistributeFees(ctx sdk.Context) {
 				// If purchaseProtectionEndTime > previousBlockTime, update service fees.
 				// Otherwise services fees were updated in the last block.
 				if entry.ProtectionEndTime.After(lastUpdateTime) && entry.ServiceFees.Native.IsAllPositive() {
+					if entry.ServiceFees.Native.IsZero() && ctx.BlockHeight() > common.UpdateHeight {
+						k.ProcessStakingPurchaseExpiration(ctx, poolPurchaser.PoolID, entry.PurchaseID, bondDenom,
+							poolPurchaser.Purchaser)
+					}
 					// Add purchaseServiceFees * (purchaseProtectionEndTime - previousBlockTime) / protectionPeriod.
```

Comment:
Should here be an else? I don't think staked shield has common processing here as purchased shield?

Reply:
Those are only for staked shield. It just adds on additional stuff that is processed in that function in addition to direct purchase, so I don't think an else branch is necessary.